lib/opam_solve.ml

```diff
@@ -296,9 +324,9 @@ module Make_solver (Context : OPAM_MONOREPO_CONTEXT) :
 
   module Solver = Opam_0install.Solver.Make (Context)
 
-  let build_context ~build_only ~allow_jbuilder ?opam_provided ?require_dune
-      ?(vendored_packages = OpamPackage.Set.empty) ~ocaml_version
-      ~local_packages ~pin_depends ~target_packages input =
+  let build_context ~build_only ~allow_jbuilder ~prefer_cross_compile
+      ?opam_provided ?require_dune ?(vendored_packages = OpamPackage.Set.empty)
```

Comment:
They are two different things. What do you mean by join them?

Reply:
Just have them next to each other in the argument.

(In fact I once tried to unify these two options with an ADT because they seem related but you get a `bool x bool` = 4 option ADT so that's not particularly helpful)

Reply:
To be honest I think we should probably drop `allows_jbuilder` as the more time passes, the less likely anyone is to use this or even for it to work with recent dune versions.